source/MaterialXGenShader/Util.cpp

```diff
@@ -839,4 +841,44 @@ void mapNodeDefToIdentiers(const std::vector<NodePtr>& nodes,
     }
 }
 
+void findLights(DocumentPtr doc, std::vector<NodePtr>& lights)
+{
+    lights.clear();
+    for (NodePtr node : doc->getNodes())
+    {
+        const TypeDesc* type = TypeDesc::get(node->getType());
+        if (type == Type::LIGHTSHADER)
+        {
+            lights.push_back(node);
+        }
+    }
+}
+
+void registerLights(DocumentPtr doc, const std::vector<NodePtr>& lights, GenContext& context)
+{
+    // Clear context light user data which is set when bindLightShader() 
+    // is called. This is necessary in case the light types have already been
+    // registered.
+    context.popUserData(HW::USER_DATA_LIGHT_SHADERS);
```

Comment:
I can change this into using the unbindLightShader interface when that is done, so we don't need to expose internal logic of the context.

Reply:
Thanks. I just wanted to make it as one place to update with this change.